### ISSUE TITLE
Can't deserialize a previously serialized `Interaction`

### ISSUE THREAD

**Issue by dthul** [serenity-rs/serenity#1398]
Serenity version: 0.10.8 with `unstable_discord_api`
Rust version: 1.53.0

I want to store `Interaction` instances in order to be able to use them at a later point in time, for example calling `Interaction::edit_original_interaction_response`. Since `Interaction` implements `Serialize` and `Deserialize` I tried using `serde_json`. Serialization works but deserialization fails with:

    Error("invalid type: null, expected struct User", line: 0, column: 0)

Not sure if this has something to do with #1164, I believe the code in question is here: https://github.com/serenity-rs/serenity/blob/5e5d92a2c5f90c88fc54fa9357fc12ae8a331041/src/model/interactions.rs#L350
It seems like it can handle the absence of the "user" field but can't handle it being `null`.

Here is the code to reproduce the failure:

`Cargo.toml`
```toml
[package]
name = "serenity-test"
version = "0.1.0"
edition = "2018"

[dependencies]
serenity = {version = "0.10", features = ["unstable_discord_api"]}
serde_json = "1.0"
```

`main.rs`
```rust
use serenity::model::interactions::Interaction;

fn main() -> Result<(), serde_json::Error> {
    let serialized = r#"{
        "id": 0,
        "application_id": 0,
        "type": 3,
        "data": {
          "custom_id": "my-component-id",
          "component_type": 2,
          "values": []
        },
        "message": {
          "id": 0,
          "attachments": [],
          "author": {
            "id": 0,
            "avatar": null,
            "bot": true,
            "discriminator": 0,
            "username": "Bot Username",
            "public_flags": 0
          },
          "channel_id": 0,
          "content": "Message content",
          "edited_timestamp": null,
          "embeds": [],
          "guild_id": null,
          "type": 0,
          "member": null,
          "mention_everyone": false,
          "mention_roles": [],
          "mention_channels": [],
          "mentions": [],
          "nonce": null,
          "pinned": false,
          "reactions": [],
          "timestamp": "2021-06-19T14:11:25.255Z",
          "tts": false,
          "webhook_id": null,
          "activity": null,
          "application": null,
          "message_reference": null,
          "flags": 0,
          "stickers": [],
          "referenced_message": null,
          "interaction": null,
          "components": [
            {
              "type": 1,
              "components": [
                {
                  "type": 2,
                  "style": 1,
                  "label": "Button Label",
                  "emoji": null,
                  "custom_id": "my-component-id",
                  "url": null,
                  "disabled": false
                }
              ]
            }
          ]
        },
        "guild_id": 0,
        "channel_id": 0,
        "member": {
          "deaf": false,
          "guild_id": 0,
          "joined_at": "2019-07-17T15:00:00.000Z",
          "mute": false,
          "nick": null,
          "roles": [0, 1, 2],
          "user": {
            "id": 0,
            "avatar": "d34db33f",
            "bot": false,
            "discriminator": 0,
            "username": "some username",
            "public_flags": 0
          },
          "pending": false,
          "premium_since": null,
          "permissions": "8589410303"
        },
        "user": null,
        "token": "S0m3sup3RL0Ngt0K3NH3rE",
        "version": 1
      }"#;
    let _: Interaction = serde_json::from_str(serialized)?;
    Ok(())
}
```

**Comment by Zaitara** [comment]
The PR #1381 will fix it.

**Comment by dthul** [comment]
Great! Didn't occur to me to check the PRs.

**Comment by Zaitara** [comment]
This can be closed as #1381 has been merged.
